modules/server/src/main/ts/bedrock/auto/Driver.ts

```diff
@@ -141,15 +141,17 @@ const focusBrowser = (browserName: string, settings: DriverSettings) => {
   }
 };
 
-const setupShutdown = (driver: webdriverio.BrowserObject, driverApi: DriverLoader.DriverAPI): (immediate?: boolean) => Promise<void> => {
+const setupShutdown = (driver: webdriverio.BrowserObject, driverApi: DriverLoader.DriverAPI, shutdownDelay: number = 0): (immediate?: boolean) => Promise<void> => {
```

Comment:
What's the interplay between `shutdownDelay` and `immediate`? Can we use the `immediate  behaviour when `shutdownDelay === 0` ?

Reply:
It's more that this is different places calling it... as the delay is set up when creating the shutdown hook, whereas immediate is done via whatever tells the driver to shutdown.

As for `shutdownDelay === 0` we wouldn't want to do that, as `deleteSession` can take some time to run and we don't want to stop the driver until that has occurred by default.

Reply:
Looking more the immediate logic isn't used anymore, so it could actually be removed.

Reply:
For reference, the immediate logic is only used when the process is forcibly killed.